consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -121,8 +122,9 @@ impl<T: Payload> ChainedBftSMR<T> {
                         idle_duration = pre_select_instant.elapsed();
                         event_processor.process_sync_info_msg(sync_info_msg.0, sync_info_msg.1).await;
                     }
-                    epoch_change = network_receivers.epoch_change.select_next_some() => {
+                    ledger_info = network_receivers.epoch_change.select_next_some() => {
                         idle_duration = pre_select_instant.elapsed();
+                        event_processor = epoch_manager.start_new_epoch(ledger_info);
```

Comment:
As far as I understand it could happen that the receiver of the epoch change is already at that (or future?) epoch. In this case we don't want to drop and replace the event processor, right?

Reply:
Yes! I'm writing another PR to filter messages based on epoch numbers, currently we assume all messages come from the same epoch.

Basically older epoch messages would be discarded, future epoch messages would be forwarded to another function to process.

Reply:
Oh, ok then. Looks good to me!
@bors-libra r+